lti_consumer/plugin/views.py

```diff
@@ -231,3 +240,153 @@ def scores(self, request, *args, **kwargs):
             status=status.HTTP_201_CREATED,
             headers=headers
         )
+
+
+class LtiNrpsViewSet(mixins.ListModelMixin, viewsets.GenericViewSet):
+    """
+    Api Endpoints for LTI NPRS Services
+    """
+    serializer_class = LtiNrpsContextMembershipSerializer
+
+    # Custom permission classes for LTI APIs
+    authentication_classes = [Lti1p3ApiAuthentication]
+    permission_classes = [LtiNrpsPermissions]
+
+    # Renderer classes to accept LTI NRPS content types
+    renderer_classes = [
+        MembershipResultRenderer,
+    ]
+
+    def get_page_size(self):
+        """
+        Get Page size for pagination
+        """
+        return settings.REST_FRAMEWORK.get('PAGE_SIZE', 10)
+
+    def build_queryset(self, filter_by=None):
+        """
+        Given a course key return a list of enrollments.
+        """
+        # pylint: disable=import-error,import-outside-toplevel,no-name-in-module
+        from common.djangoapps.student.models import CourseAccessRole
+
+        course_key = self.request.lti_configuration.location.course_key
+
+        User = get_user_model()
```

Comment:
I don't think we should use this method here. I prefer converting most of this core into a core platform API.

Reply:
https://github.com/edx/edx-platform/pull/25843 will remove this.